lib/socket-writer.js

```diff
@@ -0,0 +1,34 @@
+"use strict";
+
+var SocketIO = require("socket.io-client");
+
+// Writer for socket.io. Does not write/emit message if the socket is not connected.
+// If messages are sent and there is not a receiver, messages will be buffered
+// indefinately causing a memory leak.
+// Potentially related to: https://github.com/socketio/socket.io-client/issues/845
+module.exports = function create(options) {
+
+  var connected;
```

Comment:
This will be initialized to `undefined` which could be problematic if the socket lifecycle events don't fire.